app/renderer/actions/Inspector.js

```diff
@@ -432,23 +432,46 @@ export function searchForElement (strategy, selector) {
 }
 
 
-export function resetSearchForPORootElement () {
+export function resetSearchForPOElements () {
   return (dispatch) => {
     dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS});
   };
 }
 
-export function searchForPORootElement (strategy, selector, po) {
+export function searchForPOElements (po, strategyMap) {
+  const [type, selector] = Object.entries(po.selector)[0];
+  let strategy = strategyMap[type];
+
   return async (dispatch, getState) => {
     try {
       const callAction = callClientMethod({strategy, selector, fetchArray: true});
-      let {elements, variableName} = await callAction(dispatch, getState);
-      const findAction = findAndAssign(strategy, selector, variableName, true);
-      findAction(dispatch, getState);
-      elements = elements.map((el) => el.id);
+      let {elements} = await callAction(dispatch, getState);
       if (elements.length > 0) {
-        elements.push(po.root);
-        dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS_COMPLETED, elements});
+        let passed = true;
+        if (po.elements) {
```

Comment:
replace L451-454 into if (po_elements?.length > 0)

Reply:
will update.

Reply:
the '?.' operator does not pass the build with the error message [Unexpected token: punc (.)], replace it with "if (po.elements && po.elements.length > 0)".